src/gf.c

```diff
@@ -2316,8 +2316,10 @@ jl_code_instance_t *jl_method_compiled(jl_method_instance_t *mi, size_t world)
     jl_code_instance_t *codeinst = jl_atomic_load_relaxed(&mi->cache);
     while (codeinst) {
         if (codeinst->min_world <= world && world <= codeinst->max_world) {
-            if (jl_atomic_load_relaxed(&codeinst->invoke) != NULL)
+            jl_callptr_t invoke = jl_atomic_load_acquire(&codeinst->invoke);
```

Comment:
Why was this upgraded to an acquire load? That seems unnecessary

Reply:
This is also not NFC, since it changes the behavior of the program

Reply:
I used a safer `aquire`, aligning it with updates in similar code found in `_jl_invoke`. If it is unnecessary, we should indeed revert the changes.

Reply:
Oh I missed the atomic change here :(. I saw the relaxed load above and just interpolated .